services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,55 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(event == NULL){
+    return ERR_CODE_INVALID_ARG;
+  }
+  if(thermalMgrQueueHandle == NULL){
+    return ERR_CODE_INVALID_STATE;
+  }
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdPASS){
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  //define new event type to send to thermal manager queue 
+  thermal_mgr_event_t sendEvent;
+  sendEvent.type = THERMAL_MGR_EVENT_INTERRUPT_TRIGGERED;
+  thermalMgrSendEvent(&sendEvent);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t recieved;
+    float temp;
+    error_code_t errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```

Comment:
This should be done only after receiving an event. Imagine you read the temperature and then wait for an event. You might not receive an event for some time. So you'd be using an old value for telemetry and overtemperature detection.

Also, you should log an error right after it occurs. Right now, you're only logging a readTemp error after an event is received. This is an unnecessary delay and could lead to misleading logs.